### PR TITLE
feat: FileInput component (internal)

### PR DESCRIPTION
**Pull request by katiegeorge**
### Description

This PR is part one of multiple for file upload decomposition, to make the pull requests easier to review. 

Full plan:
- [ ] File input (internal) - **_this PR_**
- [ ] File token group (internal)
- [x] File dropzone (internal)
- [ ] PR to move all 3 out of `/internal` to make public
- [ ] Adding `file-upload` variant to button group

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/components/file-input/index.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    };
+    if (formFieldContext.invalid) {
+      nativeAttributes['aria-invalid'] = true;
+    }
+    if (ariaRequired) {
+      nativeAttributes['aria-required'] = true;
+    }
+
+    if (variant === 'icon' && !ariaLabel) {
+      warnOnce('FileInput', 'Aria label is required with icon variant.');
```
Comment (katiegeorge) [inline src/internal/components/file-input/index.tsx:72]: For reviewer: is this how we typically enforce a property that is only required in certain configurations? Should we throw an error instead?

Comment (katiegeorge) [reply src/internal/components/file-input/index.tsx:70]: Discussed with Andrei - this is typical.

File: src/internal/components/file-input/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  variant?: 'button' | 'icon';
+
+  /**
+   * Adds `aria-label` to the file input element. Use this to provide an accessible name for file inputs
+   * that don't have visible text, and to distinguish between multiple file inputs with identical visible text.
+   */
+  ariaLabel?: string;
+
+  /**
+   * Text of label (default property) - used as aria label for button variant.
```
Comment (cansuaa) [inline src/internal/components/file-input/interfaces.ts:20]: What's default property?

File: src/internal/components/file-input/index.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        }
+        uploadInputRef.current!.files = dataTransfer.files;
+      }
+      if (uploadInputRef.current) {
+        uploadInputRef.current.value = ''; // reset value to allow calling onChange when the same file is uploaded again
+      }
+    }, [value]);
+
+    return (
+      <div className={clsx(styles['file-input-container'])}>
```
Comment (cansuaa) [inline src/internal/components/file-input/index.tsx:89]: Minor: Why not root but file-input-container?

File: src/internal/components/file-input/__tests__/file-input.test.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    </div>
+  );
+  const element = renderResult.container.querySelector<HTMLElement>(`.${FileInputWrapper.rootSelector}`)!;
+  return new FileInputWrapper(element)!;
+}
+
+describe('FileInput input', () => {
+  test('`multiple` property is assigned', () => {
+    render({ multiple: false });
+    screen.debug();
```
Comment (cansuaa) [inline src/internal/components/file-input/__tests__/file-input.test.tsx:56]: Is this leftover?

File: src/internal/components/file-input/interfaces.ts (hunk 2 above)
Comment (katiegeorge) [reply src/internal/components/file-input/interfaces.ts:20]: It's the property that shows up as children... ie
```
<FileInput>{children}</FileInput
```

I am copying the [button component's](https://github.com/cloudscape-design/components/blob/main/src/button/interfaces.ts#L83) text property here.

But good call, we don't say "(default property)" specifically there, so I'll remove from the description here.

File: src/internal/components/file-input/index.tsx (hunk 3 above)
Comment (katiegeorge) [reply src/internal/components/file-input/index.tsx:89]: It's just what it was called previously. I'll update to be `root` for consistency.

File: src/internal/components/file-input/interfaces.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+   * Adds `aria-label` to the file input element. Use this to provide an accessible name for file inputs
+   * that don't have visible text, and to distinguish between multiple file inputs with identical visible text.
+   */
+  ariaLabel?: string;
+
+  /**
+   * Text displayed in the file input component. Used as the aria label for the button variant.
+   * @displayname text
+   */
+  children?: string;
```
Comment (cansuaa) [inline src/internal/components/file-input/interfaces.ts:23]: Minor: This is used as aria label if ariaLabel is not defined, not sure if we need to clarify that

File: src/internal/components/file-input/__tests__/file-input.test.tsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      <div id="test-label">Test label</div>
+    </div>
+  );
+  const element = renderResult.container.querySelector<HTMLElement>(`.${FileInputWrapper.rootSelector}`)!;
+  return new FileInputWrapper(element)!;
+}
+
+describe('FileInput input', () => {
+  test('`multiple` property is assigned', () => {
+    render({ multiple: false });
```
Comment (cansuaa) [inline src/internal/components/file-input/__tests__/file-input.test.tsx:55]: Again, is this needed? Multiple is already rendered as false below in the assertion

**Review by cansuaa — DISMISSED** [review]

File: src/internal/components/file-input/interfaces.ts (hunk 5 above)
Comment (katiegeorge) [reply src/internal/components/file-input/interfaces.ts:23]: Updated to be more clear.

**Review by cansuaa — APPROVED** [review]

**Pull request merged**